docs: add basic autosave docs (#597)
SwissDataScienceCenter/renku: docs/user/autosave.rst

.. _autosave:

Autosave in JupyterLab
======================

> Every time you stop a JupyterLab notebook, a webhook is triggered and a

Reviewer: Maybe we can simplify the phrasing here even a bit more - end users don't necessarily know what a webhook is.  Something like: 

"When you stop a JupyterLab notebook, an automatic check looks for any work that has not been pushed to the remote repository, including untracked and modified files."

scan is performed to check for any local work not pushed to the remote
repository, including untracked and modified files. If something is found,

> a new autosaved branch will be created on Gitlab.

Reviewer: ```suggestion
a new "autosave" branch will be created and pushed to GitLab.
```

> a new "autosave" branch will be created and pushed to GitLab.

Reviewer: ```suggestion
a new "autosave" branch is created and pushed to GitLab.
```

.. note::

>   The UI will filter these branches out but you will still be able to see

Reviewer: I think we can simplify a bit:

```suggestion
  You cannot start a notebook server from these branches but you can still see
```

>   them listed on Gitlab. You can easily identify them because their name

Reviewer: ```suggestion
  them listed on GitLab. You can easily identify them because their name
```

>   them listed on GitLab. You can easily identify them because their name

Reviewer: ```suggestion
  them listed in GitLab. You can easily identify them because their name
```

  starts with `renku/autosave/`

Restore unsaved work
--------------------

The next time you start a new JupyterLab server from the same branch/commit
combination, you will be notified and the autosaved data will be automatically

> loaded locally. Please note that nothing will be pushed automatically to the

Reviewer: ```suggestion
loaded in your session. Please note that nothing will be pushed automatically to the
```

origin, therefore you won't see any changes in your project's files in Renku,
nor will the :ref:`lineage` be updated.

> Local branches you created but never pushed to the origin won't be saved.

Reviewer: ```suggestion
Local branches you created but never pushed to the origin will not be saved.
```

Discard autosaved content
-------------------------

> You can easily discard the autosaved content, if you don't need it, using the

Reviewer: ```suggestion
If you don't need the autosaved content, you can easily discard it using the
```

following commands:

::

    $ git reset --hard origin/<branch-name> # remove commits and modified files
    $ git clean -f -d # remove untracked files

> If you are working on ``master`` branch, the command looks like this:

Reviewer: ```suggestion
If you are working on the ``master`` branch, the command looks like this:
```
